numpy/ma/core.py

```diff
@@ -3121,6 +3121,13 @@ def __getitem__(self, indx):
             if isinstance(indx, basestring):
                 if self._fill_value is not None:
                     dout._fill_value = self._fill_value[indx]
+                    # If we're indexing a multidimensional field in a
+                    # structured array (such as dtype("(2)i2,(2)i1")),
```

Comment:
Would be clearer is written `dtype("(2,)i2,(2,)i1")`. Current success looks like a parse oddity.